graf2d/gpadv7/src/RAttrBase.cxx

```diff
@@ -182,3 +187,35 @@ void ROOT::Experimental::RAttrBase::Clear()
    for (const auto &entry : GetDefaults())
       ClearValue(entry.first);
 }
+
+
+///////////////////////////////////////////////////////////////////////////////
+/// Collect all attributes in derived class
+/// Works only if such class has dictionary.
+/// In special cases one has to provide implementation - see RAttrColor::CollectDefaults() example
+
+ROOT::Experimental::RAttrMap ROOT::Experimental::RAttrBase::CollectDefaults() const
+{
+   ROOT::Experimental::RAttrMap res;
+
+   const std::type_info &info = typeid(*this);
+   auto thisClass = TClass::GetClass(info);
+   auto baseClass = TClass::GetClass<ROOT::Experimental::RAttrBase>();
+   if (thisClass && baseClass) {
+
+      TIter iter(thisClass->GetListOfDataMembers());
+      TObject *member = nullptr;
+      while ((member = iter()) != nullptr) {
+         TDataMember *data_member = dynamic_cast<TDataMember *> (member);
```

Comment:
```suggestion
      for (auto data_member: TRangeDynCast<TDataMember>(thisClass->GetListOfDataMembers())) {
```

Reply:
Done

Reply:
> Also, could we - in a subsequent PR - use the member name to determine the attribute title? I.e. fFooAbc => "foo_abc", fAttrBarXyz => "bar_xyz"?

https://github.com/linev/root/commit/e63b70015f1865c036ad4d998810ba41d364d8af

Implementation is simple, but super inefficient . For each instance of any attribute class each data member will try to locate - "that is my symbolic name?". And at this moment you need to access class (`const ROOT::Experimental::RAttrBase *prnt` in commit). 

Possible way of optimization - extend defaults list and store there offset of each data member. 
In this case member name can be extracted via defaults list faster - but anyway it introduces a large overhead during object construction. 

Not sure if we want this?


In contrary, list of defaults created only when used for the first time

Reply:
OK thanks, makes sense!